pallets/accounts/src/lib.rs

```diff
@@ -476,89 +523,102 @@ pub mod pallet {
-					<EVMLinkVoting<T>>::insert(&data_hash, 1);
-					return Ok(());
-				}
-			}
+			let _ = Self::vote_count_and_write(&data_hash, threshold);
 
-			// writing here at end (THE lines below only execute if the number of votes receieved by `data_hash` passes the `threshold`)
+			// writing here at end (THE lines below only execute if the number of votes received by `data_hash` passes the `threshold`)
 
-			let amount: T::Balance = amount.saturated_into();
 			let current_block_number = <frame_system::Pallet<T>>::block_number();
 
+			let lock_period: U256 =
+				data.lock_period.try_into().map_err(|_| Error::<T>::SystematicFailure)?;
+
+			let frag_amount: <T as pallet_assets::Config>::Balance = data_amount.saturated_into();
+
+			// TODO - get FRAG price from oracle
+			/* NOTE: this is assigning 20% of Tickets and NOVA. It will be replaced by a call to
+			   an oracle to retrieve the price of FRAG/USD and mint the correct amount of Tickets.
+			 */
+			let percentage_amount = Self::apply_20_percent(data_amount);
+
+			// TODO - apply mechanism of periodic vesting of NOVA
+			// NOTE: at the moment there is no mechanism of periodic vesting of NOVA.
+			let nova_amount: <T as pallet_balances::Config>::Balance = percentage_amount.saturated_into();
+			let tickets_amount: <T as pallet_assets::Config>::Balance = percentage_amount.saturated_into();
+
 			if data.lock {
 				// If FRAG tokens were locked on Ethereum
-				// ! TODO TEST
-				let locktime: u128 =
-					data.locktime.try_into().map_err(|_| Error::<T>::SystematicFailure)?;
-				let locktime: T::Moment = locktime.saturated_into();
 				let linked = <EVMLinksReverse<T>>::get(sender.clone()); // Get the Clamor Account linked with the Ethereum Account `sender`
 				if let Some(linked) = linked {
-					// If there is no linked account, shouldn't we throw an error? - 问Gio
-					let used = <FragUsage<T>>::get(linked.clone());
-					if let Some(used) = used {
-						if used > amount {
-							// Amount is the total amount of FRAG token locked on the Fragnova-owned Ethereum Smart Contract. So that means that if `amount` is smaller than `used`, there's a systemic/catastropic failure somewhere!
-							// this is bad, the user just reduced his balance so we just "slash" all the stakes
-							// reset usage counter
-							<FragUsage<T>>::remove(linked.clone()); // Remove the Clamor Account `linked` from FragUsage
-										// force dereferencing of protos and more
-							<PendingUnlinks<T>>::append(linked.clone()); // Unlink `linked` from `sender`
-						}
-					}
-				}
+					// mint Tickets for the linked user
+					<pallet_assets::Pallet<T> as Mutate<T::AccountId>>::mint_into(
+						T::TicketsAssetId::get(),
+						&linked,
+						tickets_amount)?;
+					Self::deposit_event(TicketsMinted {sender: linked.clone(), balance: tickets_amount});
+
+					// assign NOVA
+					<pallet_balances::Pallet<T> as Unbalanced<T::AccountId>>::set_balance(
+						&linked,
+						nova_amount)?;
```

Comment:
Verify first write after, tricky cos you have errors above too so you wanna do both ops first